### PR TITLE
 Remove newlines from base64digest 

### PR DESCRIPTION
**Pull request by j8r**
We very often want a SHA512 hash to be single-lined.

Also adds docs to `OpenSSL::DigestBase`

### CODE REVIEW

**Comment by yxhuvud** [comment]
> We very often want a SHA512 hash to be single-lined.

Why do you want that?

**Comment by j8r** [comment]
> Why do you want that?

Because we usually want a SHA512 hash, not having lines breaks inside this very hash.
For example, `yarn.lock`, `APKBUILD`, or even the `sha512sum` command.

What's the use case of having a multi-line hash?!

**Comment by j8r** [comment]
I never see the need of multi-line hash, anywhere, at least for SHA* and MD5

**Comment by jhass** [comment]
@yxhuvud This only changes the `Digest` classes, not the `Base64` implementation itself.

**Comment by j8r** [comment]
@yxhuvud Ok, but real world put the base64 string on a singe-line. It should be at least the default, because the most widespread.
If people really want multi-line, we may add a boolean argument then.

**Comment by j8r** [comment]
@yxhuvud Ok, but real world applications put the base64 string on a singe-line. It should be at least the default, because the most widespread.

If people really want multi-line, we may add a boolean argument then.

Another option is to remove this method, it may be fine to use `Base64.strict_encode digest` - more flexible and no confusion possible (no questions about strict or not).

**Comment by RX14** [comment]
For `Digest` purposes, `strict_encode` seems like the correct representation.

**Comment by j8r** [comment]
Any update, should we add an argument `strict : Bool = true`?

File: spec/std/openssl/digest_spec.cr [hunk 1]
```diff
@@ -2,11 +2,11 @@ require "spec"
 require "../../../src/openssl"
 
 describe OpenSSL::Digest do
-  [
+  {
```
Comment (straight-shoota) [inline spec/std/openssl/digest_spec.cr]: This should stay an array. Changing to tuple has no benefit.

File: src/openssl/digest/digest_base.cr [hunk 2]
```diff
@@ -2,36 +2,43 @@ require "base64"
...
-    def digest
-      self.clone.finish
+    # :ditto:
+    def <<(data) : Digest
+      update(data)
     end
 
-    def <<(data)
-      update(data)
+    # Clones and finish the digest.
```
Comment (straight-shoota) [inline src/openssl/digest/digest_base.cr]:
```suggestion
    # Clones and finishes the digest.
```

**Pull request merged**